File: list_Libros.cpp
#include <iostream>
#include <list>
#include "Libro.h"
using namespace std;

int main(){
    list<Libro> coleccion;
    //
    int continuar;
    do{
        // Almacenes temporales
        string tmp_string, eleccion;
        int tmp_int;
        double tmp_double;
        Libro nuevo;
        // Captura de datos
        cout << "Titulo: ";
        getline( cin, tmp_string);
        nuevo.setNombre(tmp_string);
        cout << "Autor: ";
        getline( cin, tmp_string);
        nuevo.setAutor(tmp_string);
        cout << "Fecha publicacion: ";
        cin >> tmp_int;
        cin.ignore();
        nuevo.setPublicacion(tmp_int);
        cout << "Precio compra: ";
        cin >> tmp_double;
        cin.ignore();
        nuevo.setPrecio(tmp_double);
        // Se agrega a la lista
        cout << "[i] Agregar al inicio. [f] Agregar la final | Elección: ";
        cin >> eleccion;
        if( eleccion == "i" ){
			coleccion.push_front(nuevo);
        } else {
			coleccion.push_back(nuevo);
        } 
        cout << "... .. . Libro almacenado.\n\n";
        // ------- ------- ------- Control del bucle        
        cout << "[1] Continuar [0] Finalizar.\n";
        cout << "Ingrese su eleccion= ";
        cin >> continuar;
        cin.ignore();
    }while(continuar == 1);
    // Impresión de la lista
    cout << "Extrayendo los elementos de la lista.\n";
    cout << "-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=\n";	
    while(!coleccion.empty()){
        Libro ejemplar;
        string eleccion;
		cout << "[i] Extrae del inicio. [f] Extrae del final. | Eleccion: ";
		cin >> eleccion;
        if( eleccion == "i" ){
			ejemplar= coleccion.front();			
			coleccion.pop_front();
		}else{
			ejemplar= coleccion.back();			
			coleccion.pop_back();
		}
        cout << ejemplar.getAutor() << ". " <<
				ejemplar.getNombre() << " (" <<
				ejemplar.getPublicacion() << ")\n\t$ " <<
				ejemplar.getPrecio() << "\n";     
    }
    return 0;
}

File: list_Libros.java
import java.util.Scanner;
import java.util.LinkedList;

public class list_Libros {

    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        LinkedList<Libro> lista = new LinkedList();
        //
        String continuar;
        do {
            String eleccion;
            Libro libroNuevo = new Libro();
            // ------- ------- ------- Captura
            System.out.print("Titulo del libro: ");
            libroNuevo.setNombre(scan.nextLine());
            System.out.print("Autor: ");
            libroNuevo.setAutor(scan.nextLine());
            System.out.print("Precio de compra: ");
            libroNuevo.setPrecio(scan.nextDouble());
            System.out.print("Año de publicacion: ");
            libroNuevo.setPublicacion(scan.nextInt());
            scan.nextLine();
            // ------- ------- ------- Añade a la lista
            System.out.println("[i] Agregar al inicio. [f] Agregar la final | Eleccion: ");
            eleccion = scan.nextLine();
            if (eleccion.equals("i")) {
                lista.addFirst(libroNuevo);
            } else {
                lista.addLast(libroNuevo);
            }
            // ------- ------- ------- Control del bucle
            System.out.println("[s] Agregar otro elemento. [n] Finalizar.");
            continuar = scan.nextLine();
        } while (continuar.equals("s"));

        // ------- ------- ------- Presentando
        System.out.println("Extrayendo los elementos de la lista.\n");
        while (!lista.isEmpty()) {
            String eleccion;
            Libro iterador;
            // ------- ------- ------- Extrae de la lista
            System.out.println("[i] Extrae del inicio. [f] Extrae del final | Eleccion: ");
            eleccion = scan.nextLine();
            if (eleccion.equals("i")) {
                iterador = lista.removeFirst();
            } else {
                iterador = lista.removeLast();
            }
            System.out.println(iterador.getAutor() + "(" +
                    iterador.getPublicacion() + "). " +
                    iterador.getNombre() + "\n" +
                    iterador.getPrecio());
        }
        //
        scan.close();
    }

}
